nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-sql/src/main/java/org/nuxeo/ecm/directory/sql/SQLSession.java

```diff
@@ -646,6 +673,245 @@ public DocumentModelList query(Map<String, Serializable> filter, Set<String> ful
         }
     }
 
+    @Override
+    public DocumentModelList query(QueryBuilder queryBuilder, boolean fetchReferences, boolean countTotal) {
+        if (!hasPermission(SecurityConstants.READ)) {
+            return new DocumentModelListImpl();
+        }
+        if (FieldDetector.hasField(queryBuilder.predicate(), getPasswordField())) {
+            throw new DirectoryException("Cannot filter on password");
+        }
+        if (isMultiTenant()) {
+            // filter entries on the tenantId field also
+            String tenantId = getCurrentTenantId();
+            if (!StringUtils.isEmpty(tenantId)) {
+                // TODO don't modify passed-in value
+                queryBuilder.predicate(Predicates.eq(TENANT_ID_FIELD, tenantId));
```

Comment:
Forgotten TODO ? Can be tackled by `QueryBuilder` copy constructor right ?

Reply:
Yes I didn't have the copy constructor when I wrote this comment. Will change.